app-backend/db/src/main/scala/CampaignDao.scala

```diff
@@ -91,6 +91,45 @@ object CampaignDao extends Dao[Campaign] with ObjectPermissions[Campaign] {
       params.groupQueryParameters.groupId
     ).filter(params)
       .page(page)
+      .flatMap(toWithRelated)
+
+  def toWithRelated(
```

Comment:
I don't know if this is any better necessarily and it's really more of a style preference - but you can avoid the `match` because once `.results` is a list you can `traverse` which I think is a `no-op` when the list is empty in this case:

```suggestion
def toWithRelated(
      campaignsPage: PaginatedResponse[Campaign]
  ): ConnectionIO[PaginatedResponse[Campaign.WithRelated]] = {
    val campaignsWithRelated = campaignsPage.results.toList.traverse {
      campaign =>
        for {
          labelClassGroups <- AnnotationLabelClassGroupDao.listByCampaignId(
            campaign.id
          )
          labelClassGroupsWithClasses <- labelClassGroups traverse {
            labelClassGroup =>
              AnnotationLabelClassDao.listAnnotationLabelClassByGroupId(
                labelClassGroup.id
              ) map { cls =>
                labelClassGroup.withLabelClasses(cls)
              }
          }
        } yield {
          campaign.withRelated(labelClassGroupsWithClasses)
        }
    }
    campaignsWithRelated.map { campaigns =>
      PaginatedResponse[Campaign.WithRelated](
        campaignsPage.count,
        campaignsPage.hasPrevious,
        campaignsPage.hasNext,
        campaignsPage.page,
        campaignsPage.pageSize,
        campaigns
      )
    }
```

Reply:
I like the suggestion and I bet there's a way to do this but I'm having some trouble with an `Option`:
found: `cats.free.Free[doobie.free.connection.ConnectionOp,Option[com.rasterfoundry.datamodel.PaginatedResponse[com.rasterfoundry.datamodel.AnnotationProject.WithRelated]]]`
required: `cats.free.Free[doobie.free.connection.ConnectionOp,com.rasterfoundry.datamodel.PaginatedResponse[com.rasterfoundry.datamodel.AnnotationProject.WithRelated]]`

Reply:
Ah ok - no worries - I thought I double-checked that this type-checked.